superset-frontend/src/views/datasetList/DatasetModal.tsx

```diff
@@ -0,0 +1,318 @@
+            label: s,
+          }));
+          this.setState({
+            disableSelectSchema: false,
+            schemaLoading: false,
+            schemas,
+          });
+        })
+        .catch(e => {
+          this.setState({ schemaLoading: false, schemas: [] });
+          this.props.addDangerToast(t('Error while fetching schema list'));
+          if (e) {
+            console.error(e);
+          }
+        });
+    }
+  }
+
+  fetchTables(schema: string) {
+    const { datasource } = this.state;
+
+    if (datasource && schema) {
+      this.setState(() => ({ tableLoading: true, tables: [] }));
+      const endpoint = encodeURI(
+        `/superset/tables/${datasource.value}/` +
+          `${encodeURIComponent(schema)}/undefined/false/`,
+      );
+      SupersetClient.get({ endpoint })
+        .then(({ json: tableJson = {} }) => {
+          const options = tableJson.options.map((o: option) => ({
+            value: o.value,
+            label: o.label,
+          }));
+          this.setState(() => ({
+            disableSelectTable: false,
+            tableLoading: false,
+            tables: options,
+          }));
+        })
+        .catch(e => {
+          this.setState(() => ({ tableLoading: false, tables: [] }));
+          this.props.addDangerToast(t('Error while fetching table list'));
+          if (e) {
+            console.error(e);
+          }
+        });
+    }
+  }
+
+  renderFields() {
+    const {
+      datasource,
+      datasources,
+      disableSelectSchema,
+      disableSelectTable,
+      schema,
+      schemaLoading,
+      schemas,
+      table,
+      tableLoading,
+      tables,
+    } = this.state;
+    const { onDatabaseChange, onSchemaChange, onTableChange } = this;
+
+    return (
+      <>
+        <FieldTitle>{t('datasource')}</FieldTitle>
+        <Select
+          clearable={false}
+          ignoreAccents={false}
+          name="select-datasource"
+          onChange={onDatabaseChange}
+          options={datasources}
+          placeholder={t('Select Datasource')}
+          value={datasource}
+          width={500}
+        />
+        <FieldTitle>{t('schema')}</FieldTitle>
+        <Select
+          clearable={false}
+          ignoreAccents={false}
+          isDisabled={disableSelectSchema}
+          isLoading={schemaLoading}
+          name="select-schema"
+          onChange={onSchemaChange}
+          options={schemas}
+          placeholder={t('Select Schema')}
+          value={schema}
+          width={500}
+        />
+        <FieldTitle>{t('table')}</FieldTitle>
+        <Select
+          clearable={false}
+          ignoreAccents={false}
+          isDisabled={disableSelectTable}
+          isLoading={tableLoading}
+          name="select-table"
+          onChange={onTableChange}
+          options={tables}
+          placeholder={t('Select Table')}
+          value={table}
+          width={500}
+        />
+      </>
+    );
+  }
+
+  renderTitle() {
```

Comment:
this seems a little unnecessary, maybe it's better to inline it on line 310 instead?